rct229/rulesets/ashrae9012019/section19/section19rule8.py

```diff
@@ -0,0 +1,150 @@
+    get_hvac_zone_list_w_area_dict,
+)
+from rct229.schema.config import ureg
+from rct229.utils.assertions import getattr_
+from rct229.utils.jsonpath_utils import find_all
+
+MIN_OA_CFM = 3000 * ureg("cfm")
+OCCUPANT_DENSITY_LIMIT = 0.1 / ureg("ft2")
+DEMAND_CONTROL_VENTILATION_CONTROL = schema_enums[
+    "DemandControlVentilationControlOptions"
+]
+
+
+class Section19Rule8(RuleDefinitionListIndexedBase):
+    """Rule 8 of ASHRAE 90.1-2019 Appendix G Section 19 (HVAC - General)"""
+
+    def __init__(self):
+        super(Section19Rule8, self).__init__(
+            rmrs_used=UserBaselineProposedVals(False, True, False),
+            each_rule=Section19Rule8.BuildingRule(),
+            index_rmr="baseline",
+            id="19-8",
+            description="Demand control ventilation is modeled in the baseline design in systems with outdoor air capacity greater than 3000 cfm serving areas with an average occupant design capacity greater than 100 people per 1000 ft^2.",
+            ruleset_section_title="HVAC - General",
+            standard_section=" Section G3.1.2.5 Exception #1",
+            is_primary_rule=True,
+            list_path="ruleset_model_instances[0].buildings[*]",
+        )
+
+    class BuildingRule(RuleDefinitionListIndexedBase):
+        def __init__(self):
+            super(Section19Rule8.BuildingRule, self).__init__(
+                rmrs_used=UserBaselineProposedVals(False, True, False),
+                each_rule=Section19Rule8.BuildingRule.BuildingSegmentRule(),
+                index_rmr="baseline",
+                list_path="$.building_segments[*]",
+            )
+
+        def create_data(self, context, data=None):
+            building_b = context.baseline
+            hvac_zone_list_w_area_dict_b = get_hvac_zone_list_w_area_dict(building_b)
+
+            return {"hvac_zone_list_w_area_dict_b": hvac_zone_list_w_area_dict_b}
+
+        class BuildingSegmentRule(RuleDefinitionListIndexedBase):
+            def __init__(self):
+                super(Section19Rule8.BuildingRule.BuildingSegmentRule, self).__init__(
+                    rmrs_used=UserBaselineProposedVals(False, True, False),
+                    each_rule=Section19Rule8.BuildingRule.BuildingSegmentRule.HVACRule(),
+                    index_rmr="baseline",
+                    list_path="$.heating_ventilating_air_conditioning_systems[*]",
+                    required_fields={
+                        "$": ["zones"],
+                    },
+                )
+
+            def create_data(self, context, data=None):
+                building_segment_b = context.baseline
+
+                zone_space_b = {}
+                for zone_b in find_all("$.zones[*]", building_segment_b):
+                    zone_id_b = zone_b["id"]
+                    zone_space_b[zone_id_b] = []
+                    for space_b in getattr_(zone_b, "zone", "spaces"):
+                        zone_space_b[zone_id_b].append(space_b)
```

Comment:
If the only purpose of bringing space list to the next level is to get the sum of occupants, we can do it in this function too.

```python
zone_total_occupant_dict_b = {
    zone_b["id"]: pint_sum(
                            find_all("$.spaces[*].number_of_occupants", zone_b),
                            0.0,
                        )
    for zone_b in find_all("$.zones[*]", building_segment_b)
}
```

Reply:
Great point. Addressed.